### DISCUSSION TITLE
A way to add utilities before tailwind utilities for css specificity or addUtilities before tailwind by default

### DISCUSSION THREAD

**Discussion by minimit** [tailwindlabs/tailwindcss#2667] [Ideas]
Can `addUtilities` have an argument that adds them before tailwind ones so they have less specificity? I think this should be the default behaviour at least on my use case, I tend not to override tailwind utilities but to add utilities that can be modified by tailwind ones.

Would be needed because if you add complex component utilities, you still want tailwind utilities to work on those components utilities.

For example I did a list spacing utility with list-2 with automatic margin, but i want to edit it with tailwind margins, and the margins don't work:

```
<div class="list list-2 mt-4"></div>
```

Or maybe a way to add `!important` only to some core utilities (margin).

**Comment by minimit** [comment]
Found a way to do exactly this by using `addComponents(css, variants)` instead of `addUtilities(css, variants)`

**Reply by simonswiss** [reply]
Yep - the `components` layer will be "printed" before the utilities in the generated CSS file 👍
